### PR TITLE
fix(cli): Added `run_id` and `job_name` properties to `meltano run` log messages

### PR DESCRIPTION
**Pull request by edgarrmondragon**
Quick win after a Slack thread: https://meltano.slack.com/archives/C069CQNHDNF/p1710435376837949

### CODE REVIEW

**Comment by edgarrmondragon** [comment]
Example JSON-formatted message after this change:

```json
{
    "consumer": true,
    "producer": false,
    "string_id": "target-jsonl",
    "cmd_type": "elb",
    "run_id": "bd023f5b-001b-4259-9cb2-e6abe728e902",
    "stdio": "stdout",
    "name": "target-jsonl",
    "event": "{\"bookmarks\": {\"items\": {\"replication_key\": \"time_updated\", \"replication_key_value\": 1710310147}}}",
    "level": "info",
    "timestamp": "2024-03-14T18:13:01.021353Z"
}
```

File: src/meltano/core/block/extract_load.py [hunk 1]
```diff
@@ -609,6 +609,9 @@ async def _link_io(self) -> None:  # noqa: WPS231
                 producer=block.producer,
                 string_id=block.string_id,
                 cmd_type="elb",
+                run_id=str(self.context.job.run_id)
+                if self.context.job
+                else "stateless",
```
Comment (edgarrmondragon) [inline src/meltano/core/block/extract_load.py:614]: Not sure about this. Another option is to simply omit adding `run_id` on stateless runs (i.e. those created with `--no-state-update`)

Comment (edgarrmondragon) [inline src/meltano/core/block/extract_load.py:614]:
```suggestion
                run_id=str(self.context.job.run_id) if self.context.job else None,
```

**Comment by florian-ernst-alan** [comment]
Thank you so much 🙏 Adding to this, do you think it would be possible to add the job's name? (`default:tap-postgres-to-target-snowflake`) too? This would allow for great filtering inside Datadog!

**Review by WillDaSilva — APPROVED** [review]
Approving based on the changes now, and my approval also applies if the job name is added via this PR

**Comment by edgarrmondragon** [comment]
Now with `job_name`:

```json
{
    "consumer": true,
    "producer": false,
    "string_id": "target-jsonl",
    "cmd_type": "elb",
    "run_id": "4b7ba0c5-dd27-4d8d-b286-ebb6fc1e89a6",
    "job_name": "dev:tap-getpocket-to-target-jsonl",
    "stdio": "stdout",
    "name": "target-jsonl",
    "event": "{\"bookmarks\": {\"items\": {\"replication_key\": \"time_updated\", \"replication_key_value\": 1710310147}}}",
    "level": "info",
    "timestamp": "2024-03-15T15:24:24.317836Z"
}
```

**Pull request merged**